prefect_snowflake/database.py

```diff
@@ -71,6 +71,17 @@ def _get_connect_params(self) -> Dict[str, str]:
         if connect_params.get("authenticator") == "okta_endpoint":
             connect_params["authenticator"] = connect_params.pop("okta_endpoint")
 
+        pk = connect_params.get("private_key")
+
+        if pk is not None:
+            pk = pk.decode() if isinstance(pk, bytes) else pk
+            if pk.startswith("----"):
+                from prefect_snowflake.credentials import resolve_pem_certificate
+
+                connect_params["private_key"] = resolve_pem_certificate(
+                    pk, connect_params.get("password")
+                )
```

Comment:
Perhaps this logic can be wrapped in `@pydantic.validator ` as a class method

Reply:
TBH, I was not sure if the private_key property was being used for any other purpose. If we're saying that the private_key can only be a PEM certificate, then I think I can add a validator to it. If this is the case, then I would also change the Secret to be SecretStr instead of SecretByte (but again, I don't know if this was deliberate design decision). I also don't know how a change like that may affect any existing Prefect Blocks. We can let it SecretBytes if in-doubt.
Next step for me is to understand how a Pydantic validator works!